lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -76,7 +78,7 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
   bool shutdown { false };
   
   /// The set of spawned processes to terminate if we get cancelled.
-  std::unordered_set<pid_t> spawnedProcesses;
+  std::unordered_set<sys::RedirectedProcess> spawnedProcesses;
```

Comment:
I would probably just make this a map of pid_t to `std::unique_ptr<sys::RedirectedProcess>`